lib/src/drawConfigContext.dart

```diff
@@ -0,0 +1,282 @@
+const String kRedditUrl = 'reddit_url';
+
+const String kDefaultShortUrl = 'https://redd.it';
+
+final kNotSet = null;
+
+/// The [DrawConfigContext] class provides an iterface to store.
+/// Load the DRAW's configuration file [draw.ini].
+class DrawConfigContext {
+  static final Map<String, List<String>> fieldMap = {
+    kShortUrl: [kShortUrl],
+    kCheckForUpdates: [kCheckForUpdates],
+    kKind: [kComment, kMessage, kRedditor, kSubmission, kSubReddit],
+    kOptionalField: [
+      kClientId,
+      kClientSecret,
+      kHttpProxy,
+      kHttpsProxy,
+      kRedirectUri,
+      kRefreshToken,
+      kPassword,
+      kUserAgent,
+      kUsername,
+    ],
+    kRequiredField: [kOauthUrl, kRedditUrl]
+  };
+
+  /// Path to Local, User, Global Config Files, with matching precedence.
+  Uri _localConfigPath;
+  Uri _userConfigPath;
+  Uri _globalConfigPath;
+
+  Config _customConfig;
+
+  Map<String, String> custom;
+
+  String _shortURL;
+  String _primarySiteName;
+
+  bool checkForUpdates;
+  String userAgent;
+
+  String _redirectUri;
+  String clientId;
+  String clientSecret;
+  String refreshToken;
+  String username;
+  String password;
+
+  String oauthUrl;
+  String redditUrl;
+  String redditURL;
+  String httpProxy;
+  String httpsProxy;
+
+  Uri get redirectUri => Uri.parse(_redirectUri);
+
+  //Note this Accesor throws if _shortURL is not set.
+  String get shortURL {
+    if (_shortURL == kNotSet) {
+      throw new DRAWClientException('No short domain specified');
+    }
+    return _shortURL;
+  }
+
+  /// Creates a new [DrawConfigContext] instance.
+  ///
+  /// [siteName] is the site name associated with the section of the ini file
+  /// that would like to be used to load additional configuration information.
+  /// The default behaviour is to map to the section [default].
+  ///
+  /// [userAgent] is an arbitrary identifier used by the Reddit API to diffrentiate
+  /// between client instances. Should be unqiue for example related to [siteName].
+  ///
+  /// TODO(kc3454): add ability to pass in additional prams directly.
+  DrawConfigContext({String siteName = 'default', String userAgent}) {
+    // Give passed in values highest precedence for assignemnt.
+    _primarySiteName = siteName;
+    this.userAgent = userAgent ?? kNotSet;
+    // Initialize Paths.
+    _localConfigPath = _getLocalConfigPath();
+    _userConfigPath = _getUserConfigPath();
+    _globalConfigPath = _getGlobalConfigPath();
+    // Load the first file found in order of path preference.
+    final primaryFile = _loadCorrectFile();
+    _customConfig = new Config.fromStrings(primaryFile.readAsLinesSync());
+    // Load values found in the ini file, into the object fields. 
+    fieldMap.forEach((key, value) => _fieldInitializer(key, value));
+  }
+
+  /// Check for the existence of the [primaryFile].
+  bool _checkForExistance(final primaryFile) {
+    bool fileLoaded;
+    final fileExistance = primaryFile.exists();
```

Comment:
Change this to `primaryFile.existsSync()`, otherwise I'm pretty sure this function could return `false` if the Future hasn't completed before returning.